docs/src/main/asciidoc/spring-di.adoc

```diff
@@ -1,42 +1,91 @@
+This guide explains how your {ProductName} application can leverage Dependency Injection annotations provided by the Spring Framework.
 
+ifndef::product-docs[]
+
+This guide is maintained in the main {ProductName} repository and pull requests should be submitted there:
+https://github.com/quarkusio/quarkus/tree/master/docs/src/main/asciidoc
 
 == Solution
 
-We recommend that you follow the instructions in the next sections and create the application step by step.
-However, you can go right to the completed example.
+Follow the instructions in the next sections and create the application step by step.
+Alternatively, you can clone and use the completed example.
 
 Clone the Git repository: `git clone {quickstarts-clone-url}`, or download an {quickstarts-archive-url}[archive].
 
 The solution is located in the `spring-di-quickstart` {quickstarts-tree-url}/spring-di-quickstart[directory].
+endif::[]
 
-== Creating the Maven project
-
-First, we need a new project. Create a new project with the following command:
-
+// Tech preview info for community purposes, different from product
+ifndef::product-docs[]
+include::./status-include.adoc[]
+:extension-status: preview
+endif::[]
+
+//tech preview info for product docs
+ifdef::product-docs[]
+NOTE: The `quarkus-spring-di` extension is available as link:https://access.redhat.com/support/offerings/techpreview[Technology Preview] with {ProductLongName}.
+endif::[]
+
+
+//ifdef::product-docs[]
+[IMPORTANT]
+====
+The Spring compatibility layer that {ProductName} provides does not start a Spring Application Context or execute any infrastructure classes that are provided by Spring (for example, `org.springframework.beans.factory.config.BeanPostProcessor`) when you start your application.
+{ProductName} can only read metadata from Spring classes and annotations and parse user code method return types and parameter types that are specific to Spring.
+However, when you add arbitrary libraries that are part of Spring Framework to your {ProductName} application, such libraries will not function properly, because {ProductName} is not able to use them.
+====
+//endif::[]
+
+.Prerequisites
+
+// comment out the following prerequisite in the alternative 1 layout of the document.
+ifdef::product-docs[]
+* Have a {ProductName} Maven project.
+endif::[]
+ifndef::product-docs[]
+* Have access to an IDE
+endif::[]
+* Have OpenJDK 8 installed.
+* Have Maven {maven-version} installed.
+
+ifndef::product-docs[]
+// Alternative 1: generate project form scratch
+[id="creating-the-spring-di-example-maven-project_{context}"]
+== Creating the Spring DI example Maven project
+
+[role="_abstract"]
+You can create new a new Maven project for the Spring DI example application on the command line.
+
+.Procedure
+
+. Enter the following commands to:
+
+* generate the directory structure of the project.
```

Comment:
Is it intentionally we add tons of text upfront the code to copy and play with vs have it afterward where one can refer back to it ?

Reply:
I made this change for the purposes of consistency with  product docs. Essentially to fit the format of how we document individual use case scenarios in product docs, where we typically start from the point of modifying an existing project. However, I am open to changing this section, because I consider the one-liner project setup workflow to be both appealing and practical. I suggest that we discuss this change further before we commit to a final version.